### PR TITLE
Add known orientations to dials.stills_process

### PR DESCRIPTION
**Pull request by phyy-nx**
In [Brewster 2018 (see figure 5)](https://journals.iucr.org/d/issues/2018/09/00/lp5037/lp5037fig5.html), two detectors were used to collect data: a front CSPAD detector at a standard position and a second CSPAD 2.5m behind the front detector, used to record low resolution data.  The crystal orientations from the front detector were used to index spots on the back detector using bookkeeping code originally in [`cctbx.xfel.xtc_process`](https://github.com/cctbx/cctbx_project/blob/master/xfel/command_line/xtc_process.py#L1180-L1189).  This PR ports that bookkeeping into dials.stills_process as a general use case.  A version of the referenced branch was successfully used in a recent XFEL experiment with a front and a back detector.

Specifically, this PR adds `known_orientations` as a parameter to dials.stills_process, used to specify the path to previous processing results, including crystal orientations. If specified, the known orientations for the images will be used to index the images. Provide paths to experiment list files, using wildcards as needed.  The program will load the experiment lists files up front and create a dictionary mapping filename/image number to experiment lists (including potentially multiple lattices).  Then, for each image, it refers to this dictionary and if the image had been previously indexed, it uses that orientation to index the image.

Additionally, the `require_known_orientation=False` parameter is added.   If known_orientations are provided, and an orientation for an image is not found, this is whether or not to attempt to index the image from scratch using indexing.method

Includes a test of known_orientations in `test_stills_process.py`

Note the XFEL GUI already [knows](https://github.com/cctbx/cctbx_project/blob/master/xfel/ui/db/job.py#L163-L174) about this parameter.

Co-authored-by: Daniel Paley <dwpaley@lbl.gov>

### CODE REVIEW

**Comment by phyy-nx** [comment]
The new test requires dials/data-files#42 to be merged to add a set of crystal orientations the test uses.  After merging that, it needs to be referenced in the image_examples dataset in dials-data.

**Comment by phyy-nx** [comment]
Hi, I'd like to merge this tomorrow unless there is feedback.  Thanks!

File: src/dials/command_line/stills_process.py
Comment (dwpaley) [inline src/dials/command_line/stills_process.py:1174]:
```suggestion
                indexing_succeeded = True
```

**Review by dwpaley — CHANGES_REQUESTED** [review]

File: tests/command_line/test_stills_process.py
Comment (phyy-nx) [inline tests/command_line/test_stills_process.py:146]: Need to uncomment this and be sure it passes, meaning dials.data  should include `SACLA-MPCCD-run266702-0-subset-known_orientations.expt`

**Review by dwpaley — APPROVED** [review]

**Comment by phyy-nx** [comment]
Test failure seems unrelated and is on main.  Merging.  Thanks @dwpaley.

**Pull request merged**
